metapool/controls.py

```diff
@@ -53,6 +59,39 @@ def get_all_projects_in_context(sample_context):
     return study_ids
 
 
+def get_controls_details_from_context(sample_context):
+    if sample_context is None:
+        return {}
+
+    result = {}
+    # convert the sample context dataframe to a list of dictionaries
+    controls_dicts_list = sample_context.to_dict(orient="records")
+    for curr_dict in controls_dicts_list:
+        curr_sample_name = curr_dict[SAMPLE_NAME_KEY]
+        curr_dict[SECONDARY_STUDIES_KEY] = sorted(_split_secondary_studies(
+            curr_dict[SECONDARY_STUDIES_KEY]))
+        result[curr_sample_name] = curr_dict
+    return result
+
+
+def make_manual_control_details(sample_name, primary_study,
+                                secondary_studies=None, sample_type=None):
+
+    if sample_type is None:
+        sample_type = BLANK_SAMPLE_TYPE
+
+    if secondary_studies is None:
+        secondary_studies = []
+
+    details_dict = {
```

Comment:
@charles-cowart sorry if I'm being braindead ... can you clarify "a defined time"?  Are you envisioning that we define a class or struct or something for this data structure rather than just a (mutable) dictionary?

Reply:
I'm so sorry, that must have been autocorrect - I meant to write 'a defined type'.